docs/DeveloperGuide.md

```diff
@@ -1,29 +1,194 @@
+organization and tracking. The implementation of this feature involves interactions across multiple classes within the 
+system. 
+
+#### Overview
+The process of categorizing books by genre is a multistep operation that involves the following classes:
+1. `ParserGenre`: This class contains methods that handle the categorization of books.
+2. `BookGenre`: Individual book objects are updated with their respective genres directly in this class.
+3. `ParserMain`: This class is responsible for parsing the input command to extract the specific index and genre.
+
+#### Architecture-Level Design:
+
+- The book management system is structured using a layered architecture model, comprising the UI layer, the command 
+parser layer, and the data model layer.
+- The `set-genre` feature is primarily situated in the command parser layer but interacts closely with the data model 
+layer to update the genre of a book.
+
+#### Component-Level Design:
+
+- `UI Layer`: Receives input from the user and displays the results of operations. It's where the user's request to set 
+  a genre starts and ends with feedback.
+- `ParserMain` (Command Parser Layer): Acts as the central hub for command processing, determining the type of command 
+  and delegating to the specific parser.
+- `ParserGenre` (Command Parser Layer): Specializes in handling the set-genre command. It prompts the user for inputs 
+  and validates them.
+- `BookList` (Data Model Layer): Maintains the list of books and genres. It's queried to ensure the book index is valid 
+  and to retrieve or update the list of genres.
+- `BookGenre` (Data Model Layer): Provides the functionality to set the genre of a book in the `BookList`.
+
+#### Implementation Details:
+
+- Upon invoking the `set-genre` command, `ParserMain` interprets the command and forwards it to `ParserGenre`.
+ `ParserGenre` then guides the user through selecting an existing genre or adding a new one, ensuring valid inputs at 
+  each step.
+- The chosen genre is then applied to the specified book through `BookGenre`, which interfaces with `BookList` to make 
+  the update.
+- Throughout this process, `Ui` is called upon to display messages, guiding the user and confirming the successful 
+  update.
+
+#### Rationale for Design:
+
+- The command pattern used allows for easy addition of new commands and features without altering existing code 
+  structures, adhering to open/closed principles.
+- The clear separation of concerns makes the system robust, with each component focusing on a single responsibility, 
+  enhancing maintainability and scalability.
+
+#### Alternatives Considered:
+
+- An embedded approach, where genre setting logic is part of a larger class managing all book attributes, was 
+  considered. However, this was rejected due to potential scalability issues and difficulty in maintaining code.
+
+![SetGenreSequenceDiagram.png](UML_diagrams/SetGenreSequenceDiagram.png)
```

Comment:
![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/99406560/dba9a645-b442-48c2-9603-bd4a7657c6fc)

Is this correct? Should it be returning any value(eg genres)?